### PR TITLE
Update six to support Python 3.11

### PR DESCRIPTION
**Pull request by gpotter2**
- Updates `six` to support `Python 3.10` & `Python 3.11`
- moves `six` to `libs/`. It made no sense for it to be in `modules`
- remove some six imports when not needed

fixes https://github.com/secdev/scapy/issues/3502

### CODE REVIEW

File: scapy/layers/inet.py [hunk 1]
```diff
@@ -2076,6 +2075,7 @@ def fragleak2(target, timeout=0.4, onlyasc=0, count=None):
 class ICMPEcho_am(AnsweringMachine):
     """Responds to ICMP Echo-Requests (ping)"""
     function_name = "icmpechod"
+    send_function = staticmethod(sendp)
```
Comment (guedou) [inline scapy/layers/inet.py:2078]: Why did you change this?

**Review by guedou — DISMISSED** [review]
LGTM. I did not review changes to six.py Do you want me to read it?

File: scapy/layers/inet.py (hunk 1 above)
Comment (gpotter2) [reply scapy/layers/inet.py:2078]: Yeah this has nothing to do with this PR, good catch

**Review by guedou — APPROVED** [review]

**Comment by guedou** [comment]
Do you want me to merge it, or do you need to modify more things?

**Comment by gpotter2** [comment]
Should be ready to merge.

I'd like to make 2.5.0 quite soon, what do you think? We have typing for the entire core already (minus arch/), and it fixes issues with OSx, new python versions...

**Comment by guedou** [comment]
We could start to make the usual Kansan project and sort PRs. I will do my best to be more available for Scapy in the next months.

What is the rationale to release without typing everything?

**Comment by gpotter2** [comment]
Typing everything is out of scope (way too long). Typing the core is already very useful to anyone developing a module using Scapy.

**Comment by guedou** [comment]
Thanks! That makes perfect sense.
